### ISSUE TITLE
Data Exchange Advanced Search Feature Request

### ISSUE THREAD

**Issue by shelbysawyer** [Riverscapes/RiverscapesXML#704]
Posting in this repo because I don't have access to the data exchange repo--

I created several Collections in the Data Exchange to pair with figures in my thesis (and, later, a manuscript) that feature Riverscapes projects. The manual creation of Collections worked well when I only needed a few projects (e.g., <5), but it wasn't worth the effort required for figures that used projects from whole 2-digit (region), 4-digit (subregion) or 6-digit (basin) HUCs.

For context:
1. the average region (2-digit HUC) has 864 10-digit HUCs
2. the average subregion (4-digit HUC) has 76 10-digit HUCs
3. the average basin (6-digit HUC) has 47 10-digit HUCs
5. the average subbasin (8-digit HUC) has 8 10-digit HUCs

**This got me thinking: it would be handy if the Advanced Search option to specify by HUC wasn't limited only to exact 10-digit HUC matches.** 

With this feature, instead of manually searching for and adding the roughly 47 projects in a basin, we could type in the 6-digit HUC code, project type, and version to return all 47 projects with the same first 6 digits of their HUC code in one search. 

**This is where part 2 of my request comes in-- could we find a way to add all projects that fit search criteria to a collection in one go?** 

I know the 1,000 project limit would still apply to collections, but considering that the average 2-digit HUC (the biggest) has an average of 864, my guess is that it wouldn't be that common. Also, I'd think a simple message popup reading something like "This search criteria exceeds the maximum number of projects allowed (1,000) in a collection. Try refining your search by project type, version, and/or extent" would be fine. 

**In summary, the ability to 1) identify projects using 2-, 4-, 6-, and 8-digit HUC codes and 2) add all projects (<1,000) that fit given search criteria to a collection at once would be very useful for transparency and citation purposes.**

**Comment by MattReimer** [comment]
This is a good suggestion @shelbysawyer. I can totally see the use case too. I need to think about it a little more broadly because HUC is a US-centric measure and we need this to work everywhere but it definitely informs how we could inform search going forward.

For example, there's nothing but a little effort stopping us from implementing a "starts-with", "ends-with" and "is exactly" dropdown on metadata searching.

Ill definitely bring this up the next time there's a phase of work on the data exchange

**Comment by MattReimer** [comment]
The search engine has had some dramatic improvements. We could still do more but I'm closing this ticket as I believe a lot of it has been addressed.
